packages/core/src/Repo.ts

```diff
@@ -276,6 +253,30 @@ export class Repo<T = any> extends EventEmitter {
     return newDoc
   }
 
+  /**
+   * Used for sending the entire current state of the repo to a new peer.
+   * @returns Returns an object mapping documentIds to an array of changes.
+   */
+  async getHistory(): Promise<RepoHistory> {
+    const history: RepoHistory = {}
+    // TODO: for large datasets, send in batches
+    for await (const cursor of this.storage.changes) {
+      const { documentId, changes } = cursor.value as ChangeSet
```

Comment:
We won't need this cast once we fix the type of `changes`

Reply:
👍 58de7e1ec3557b21262987bd681db3f7ad9ef74e